### DISCUSSION TITLE
Issues upgrading to 3.4

### DISCUSSION THREAD

**Discussion by jsmcnair** [sse-secure-systems/connaisseur#1594] [Q&A]
Hi.

I had a few issues migrating to v3.4 and wanted to share my experience. I can submit fixes for these if necessary.

We are using DCT validators against private containers. The first issue I experienced was that the file name required for the repository credentials has change to `secret.yaml` but this has not yet been reflected in the documentation: https://github.com/sse-secure-systems/connaisseur/blob/6ca19ae281c46a441ab9feb2c39f292b7e402e12/docs/validators/notaryv1.md?plain=1#L188

Also when providing the public key I did not update value to include the headers (e.g. `BEGIN PUBLIC KEY`), which I found caused a panic in connaisseur when trying to decode the trust root key. I think it's because the the `pem.Decode` method returns `nil` if it doesn't find a key so `x509.ParsePKIXPublicKey` gets passed a nil pointer: https://github.com/sse-secure-systems/connaisseur/blob/6ca19ae281c46a441ab9feb2c39f292b7e402e12/internal/validator/notaryv1/notaryv1_validator.go#L217. Once I updated the key format being passed to the application configuration the nil pointer panic went away, but I feel this should be checked for and properly handled in the code.

**Comment by phbelitz** [comment]
thank you for your feedback!

I created a PR to fix this: https://github.com/sse-secure-systems/connaisseur/pull/1624
